internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs

```diff
@@ -1215,10 +1235,109 @@ mod tests {
             .unwrap();
 
         repository
-            .remove_rolled_back_transactions_and_block_range(BlockRange::LENGTH * 3)
+            .remove_rolled_back_transactions_and_block_range_by_block_number(BlockRange::LENGTH * 3)
             .await
             .unwrap();
         assert_eq!(2, repository.get_all_transactions().await.unwrap().len());
         assert_eq!(2, repository.get_all_block_range_root().unwrap().len());
     }
+
+    #[tokio::test]
+    async fn remove_rolled_back_transactions_and_block_range_by_slot_number() {
```

Comment:
It might be a bit of a stretch, but would it make sense to add a case the target block and the blocks above contain multiple transactions? To ensure that only the transactions from the blocks above the target block are removed.